include/cocaine/rpc/asio/readable_stream.hpp

```diff
@@ -55,6 +56,8 @@ class readable_stream:
 
     decoder_type m_decoder;
 
+    synchronized<bool> m_stopped{false};
```

Comment:
Could be generic `std::atomic<bool>` more  convenient? The same for `writable_stream`.

Reply:
`std::atomic` doesn't have RAII scope lock.